### ISSUE TITLE
3D Poly Line Model - Can't resize drops.

### ISSUE THREAD

**Issue by neowolfwitch** [smeighan/xLights#3688]
MacOS Ventura, xLights v. 2023-01

If you add drops to the Poly Line model (for Peace/Pixel Stakes) you cannot adjust their height reliably, and trying to do so yields strange results. You can resize on a slant, but not if the line is horizontal. Sometimes resizing causes the line to way below or way above the ground plane. I'm attaching a video link that hopefully makes it clearer.

**To Reproduce**
To reproduce- all you need to do is create a 3D poly line with drops and attempt to resize the height.

**Expected behavior**
You should be able to adjust the height of the drops, either using the stretch manipulation handle, or by using the Dimensions height property (which doesn't exist right now).

**Versions (please complete the following information):**
Noted above: MacOS Ventura, xLights v. 2023-01

Video link: https://youtu.be/_QaUEWmhe_4

**Comment by neowolfwitch** [comment]
I noticed CTRL+Z a few times in the video when I re-watched it. That's a throwback to my old Windows days. On Mac it's obviously CMD+Z.

**Comment by AzGilrock** [comment]
Yeah I can't remember if I even provided a method to lengthen the icicles for PolyLine.  Using the Scale tool is not the right way that is going to attempt to scale the entire model not just the icicles and the drops do not really count towards the height of the model so that's why scaling a horizontal line doesn't work since its multiplying against zero.  There are definitely scaling problems with the PolyLine in general that have nothing to do with the icicle drops.  I was working on it last year and did not have a fix for it yet and meant to get back to it.  It's not common to need to scale a PolyLine so nobody has mentioned it before.

**Comment by neowolfwitch** [comment]
Thanks @AzGilrock.

Someone recommended using the Candy Cane model and setting it up as "Stick" as a better way of doing stakes. That's working very well for what I need to do, except for the 20 cane limit that seems to be enforced.

It seems like Poly Line with drops might not be the best way to do this, since the Icicle model already has similar functionality. I know slanting the stakes isn't a big deal, but I try to make my 3D model as close to the real thing as possible, as I figure that will also yield the best render results. Probably nothing anyone but me would notice though. 😊

I think what would really help some of us out is to have something similar to the cube model, but allow it to be mapped similar to a terrain, or- a purpose-built Stake model. a cube works great if you have a level, or at least evenly sloped, yard, but not if you are trying to work it into an odd-shaped hill.

**Comment by AzGilrock** [comment]
There is also some code Keith put in PolyLine awhile back where if there are only 2 points with no curve it sets the scale to 0 which prevents it from scaling.  The goal was to fix an issue where the PolyLine will draw at a 45 degree angle in the Model Preview window.  Its not the best fix but I worked on it and couldn't fix it for all situations so I left it as is.

**Comment by cybercop23** [comment]
Gents, is this still a thing likely to get changed/implemented?
